test/cpp/lazy/test_lazy_ops.cpp

```diff
@@ -7311,6 +7311,26 @@ TEST_F(LazyOpsTest, TestNarrowCopy) {
   }
 }
 
+TEST_F(LazyOpsTest, NonZeroNarrowCopy) {
+  FLAGS_ltc_enable_symbolic_shapes = true;
+  auto x_cpu = torch::rand({10});
+  auto y_cpu = torch::nonzero(x_cpu);
+  int64_t y0_cpu_size = y_cpu.sizes()[0];
+  auto b_cpu =
+      torch::randn({10}); // a base tensor so we can compute with symint
+  auto z_cpu = b_cpu.narrow_copy(0, 0, y0_cpu_size);
+
+  // Same operations, but on LAZY tensor
+  auto x_lazy = x_cpu.to(kLazy);
+  auto y_lazy = torch::nonzero(x_lazy);
+  // y_lazy is a lazy tensor with the upper bound x_lazy.size(0)
+  SymInt y0_lazy_size = y_lazy.sym_sizes()[0];
+  auto b_lazy = b_cpu.to(kLazy);
+  auto z_lazy = b_lazy.narrow_copy_symint(0, 0, y0_lazy_size);
+  ASSERT_EQ(y_lazy.sizes()[0], 10);
```

Comment:
is this checking the upper bound?

Reply:
correct, this is the test we included in https://docs.google.com/document/d/1qUBwlLbDPsWKP3kaSCEgIF4hf-C4vvpA10upEvZep_s/edit# p31